src/mock/MockNonPermitNonERC20WithDS.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+contract MockNonPermitNonERC20WithDS {
```

Comment:
Do you really need to copy-paste this contract from the `test/mocks/MockNonPermitNonERC20WithDS.sol`?

Reply:
Unfortunately, hardhat does not allow using multiple paths for contracts at the same time, which causes compatibility issues with foundry. You can get more information by following this link: https://ethereum.stackexchange.com/questions/139531/hardhat-multiple-folders-for-compilation